narwhals/dataframe.py

```diff
@@ -610,6 +610,68 @@ def from_dict(
         )
         raise ValueError(msg)
 
+    @classmethod
+    def from_dicts(
+        cls,
+        data: Sequence[Mapping[str, Any]],
+        schema: IntoSchema | None = None,
+        *,
+        backend: IntoBackend[EagerAllowed],
+    ) -> DataFrame[Any]:
+        """Instantiate DataFrame from a sequence of dictionaries representing rows.
+
+        Notes:
+            For pandas-like dataframes, conversion to schema is applied after dataframe
+            creation.
+
```

Comment:
Hi @dangotbanned. This is what I had written. What do you think?

```suggestion
        When a schema is not specified, the native libraries infer the schema differently:
        - pandas uses all rows
        - polars uses the first 100 rows
        - pyarrow uses only the first row
        If the rows in `data` have different keys, consider passing a schema for more consistent
        results

```